src/shared/infra/http/controllers/wallet/wallet.controller.ts

```diff
@@ -29,4 +38,28 @@ export class WalletController {
   async createWallet(@Body() walletData: SaveWalletDTO): Promise<Wallet> {
     return this.walletUseCase.createWallet(walletData);
   }
+
+  @Get('all')
+  @ApiOkResponse(API_RESPONSES.OK)
+  @ApiNotFoundResponse(API_RESPONSES.NOT_FOUND)
+  @ApiOperation({
+    summary: 'Listar todas as carteiras.',
+    description: 'Esta rota permite visualizar todas as carteiras.',
+  })
+  async listAllWallets(): Promise<Wallet[]> {
+    const { wallets } = await this.findAllWallets.execute();
+    return wallets;
+  }
+
+  @Get(':id')
+  @ApiOperation({
+    summary: 'Listar uma carteira pelo ID.',
+    description: 'Esta rota permite visualizar os dados de uma carteira.',
+  })
+  @ApiResponse(API_RESPONSES.OK)
+  @ApiNotFoundResponse(API_RESPONSES.NOT_FOUND)
+  async listWallet(@Param('id') wallet_id: string): Promise<Wallet> {
+    const { wallet } = await this.findWalletById.execute(wallet_id);
```

Comment:
aqui por que não dar só um return this.findWalletById.execute(wallet_id); ?

Reply:
Eu tentei aqui Levi, mas ficou dependente da desestruturação para retornar apenas o wallet_id